web3/_utils/encoding.py

```diff
@@ -290,9 +290,11 @@ def encode_single_packed(_type: TypeStr, value: Any) -> bytes:
 class Web3JsonEncoder(json.JSONEncoder):
     def default(self, obj: Any) -> Union[Dict[Any, Any], HexStr]:
         if isinstance(obj, AttributeDict):
-            return {k: v for k, v in obj.items()}
-        if isinstance(obj, HexBytes):
+            return obj.__dict__
+        elif isinstance(obj, HexBytes):
             return HexStr(obj.hex())
+        elif isinstance(obj, bytes):
+            return to_hex(obj)
```

Comment:
Since you have the arg type as `Any`, I would expect an `else` clause to catch anything that isn't `AttributeDict`, `Hexbytes`, or `bytes`. Or just update the arg type to a union of those.

Reply:
I'm not sure I follow. There is no `else` but we return the default `JSONEncoder.default` if it's none of the cases on line `298`. So in that sense we only catch cases unique to *web3.py* and the rest get put through the default encoder for the `json` library.

Reply:
Nevermind, `default` is implemented correctly per the `JSONEncoder.default` docstring. All good.